src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -220,19 +222,37 @@ public Visitable visit(MethodDeclaration method, Void p) {
   @Override
   public Visitable visit(Parameter para, Void p) {
     if (insideTargetMethod) {
-      ResolvedType paraType = para.resolve().getType();
-      if (paraType.isReferenceType()) {
-        String paraTypeFullName =
-            paraType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
-        usedClass.add(paraTypeFullName);
-        for (ResolvedType typeParameterValue : paraType.asReferenceType().typeParametersValues()) {
-          String typeParameterValueName = typeParameterValue.describe();
-          if (typeParameterValueName.contains("<")) {
-            // removing the "<...>" part if there is any.
-            typeParameterValueName =
-                typeParameterValueName.substring(0, typeParameterValueName.indexOf("<"));
+      Type type = para.getType();
+      String paraTypeFullName = "";
+      ResolvedType paramType;
+      if (type.isUnionType()) {
+        for (ReferenceType param : para.getType().asUnionType().getElements()) {
+          paramType = param.resolve();
+          paraTypeFullName =
+              paramType.asReferenceType().getTypeDeclaration().get().getQualifiedName();
+          usedClass.add(paraTypeFullName);
+        }
+      } else {
+        if (para.getParentNode().isPresent() && para.getParentNode().get() instanceof CatchClause) {
```

Comment:
This `if` statement must be explained with a comment. Ideally, that comment will link to the JavaParser bug/discussion that you and I talked about in our meeting yesterday @tahiat - you said you'd asked the JavaParser people about this, and they told you it should work (but it doesn't). Substantiate and explain that claim here where you work around it in a comment.

Reply:
added comments here. 
parameter resolution works on method and construction but not on catch clause.